nni/compression/pytorch/compressor.py

```diff
@@ -580,10 +580,15 @@ class QuantType:
     """
     Enum class for quantization type.
     """
-    QUANT_INPUT = 'input'
-    QUANT_WEIGHT = 'weight'
-    QUANT_OUTPUT = 'output'
+    QUANT_INPUT = 0
+    QUANT_WEIGHT = 1
+    QUANT_OUTPUT = 2
 
+QType_Dict = {
+    0: "input",
+    1: "weight",
+    2: "output"
+}
 
 class QuantGrad(torch.autograd.Function):
```

Comment:
I think the best way to accomplish the quantization gradient calculation is to add `quant_weight`, `quant_input`, `quant_output` method to `quantizer` class and `QuantGrad` class only calls this methods instead.

Reply:
Agree, this will take us time to discuss a new design and `quantizer` may be refactored in another pr. In this pr, we only fix dorefa and bnn unfit problem.